packages/select/src/select.vue

```diff
@@ -528,6 +529,7 @@
 
       handleFocus(event) {
         if (!this.softFocus) {
+          setTimeout(() => { if (this.automaticDropdown) this.visible = true; }, 300);
```

Comment:
Why 300ms timeout? Why not set `visible` straightway?

Reply:
The `automaticDropdown` check is on a timeout, because on `mousedown` the **`input`** element will immediately emit an **`onfocus` event** that bubbles to the **`select`** component and fires `handleFocus`. The  `automaticDropdown` check is inside `handleFocus` and therefore fires on focus which happens either `onmousedown` (which is when `onfocus` is triggered) or after a tab operation lands on the element. On the other hand, `toggleMenu`, which checks if `!this.visible`,  fires ***after*** `onclick` (and therefore always after `handleFocus`). 

If  the `automaticDropdown` check fires too soon, then `this.visible === true` by the time `toggleMenu` hits, and the menu will always remain closed. By setting a timeout of 300ms the check waits long enough to fire *after* `onclick`--*in the majority of cases*; but not too long that it feels totally unresponsive.

I originally wanted to make the `automaticDropdown` check more intelligent by checking to see if `handleFocus` was invoked by an `onmousedown` event, but that's not possible because`handleFocus` is bound to a `onfocus` event which fires as a result of `onmousedown`

This discussion also answers your questions regarding the timeout on the test. To properly test this feature we have to wait the 300 ms for it to activate.

Reply:
Personally, I'm not a big fan of timeout. Timing issues is not really something we want in our code. It's hard to debug if problem arises, and may lead to flaky tests.

My idea is to add a flag to record whether `automaticDropdown` just set `visible` to `true`; Check that flag in `toggleMenu`; Then clear the flag when `visible` state changes. This hack appears to be working on simple cases.

@Leopoldthecoder Any better ideas?

Reply:
I agree with @wacky6 .

- After setting `visible` to true in this `if` block, also set a flag to `true`
- In `toggleMenu`, check this flag, and do the actual toggling only when the flag is `false`
- Reset this flag to `false` in the `visible` watcher

Also, please remember to reset `softFocus` to `false` in `handleBlur`.

Reply:
Ok. Will do. Will update soon.

Reply:
@Seebiscuit Hi, please update this PR based on discussions here.

Reply:
Yea, I'm sorry, I overlooked @wacky6 comment. The update was made in the last commit.